sli/tokenutils.cc

```diff
@@ -73,14 +73,23 @@ template <>
 double
 getValue< double >( const Token& t )
 {
-  DoubleDatum* id = dynamic_cast< DoubleDatum* >( t.datum() );
-  if ( id == NULL )
-  { // we have to create a Datum object to get the name...
-    DoubleDatum const d;
-    throw TypeMismatch( d.gettypename().toString(), t.datum()->gettypename().toString() );
+
+  DoubleDatum* dd = dynamic_cast< DoubleDatum* >( t.datum() );
+  if ( dd )
+  {
+    return dd->get();
   }
-  return id->get();
+  IntegerDatum* id = dynamic_cast< IntegerDatum* >( t.datum() );
+  if ( id )
+  {
+    return ( double ) id->get();
+  }
+
+  // we have to create a Datum object to get the name...
```

Comment:
This is a bit nitpicky, but could you use capital W for this and the other comments?

Reply:
I addressed this in b1a81009d4d0115d5debb601e1ff9b175fc28ae5.